File: src/routes/tokenClassification.ts
import { Router, type Request, type Response } from "express";
import { pipelineFactory } from "../utils/pipelineFactory";
import type {
  TokenClassificationRequest,
  PipelineResult,
} from "../types/index";

const router = Router();

router.post("/", async (req: Request, res: Response) => {
  try {
    const { text }: TokenClassificationRequest = req.body;

    if (!text) {
      return res.status(400).json({ error: "Missing required field: text" });
    }

    const pipe = await pipelineFactory.getPipeline("token-classification");
    const result = await pipe(text);

    const response: PipelineResult = {
      success: true,
      data: result,
    };

    res.json(response);
  } catch (error: any) {
    res.status(500).json({
      success: false,
      error: error.message || "Token classification failed",
    });
  }
});

export default router;

File: python-src/routes/token_classification.py
from fastapi import APIRouter
from models.types import TokenClassificationRequest, PipelineResult
from utils.pipeline_factory import pipeline_factory

router = APIRouter()

@router.post("/", response_model=PipelineResult)
async def classify_tokens(request: TokenClassificationRequest):
    # TODO: Implement token classification
    return PipelineResult(success=True, data={"tokens": "Not implemented yet"})